### DISCUSSION TITLE
Infer Skeleton size from component it replaces/overlays

### DISCUSSION THREAD

**Discussion by oliverunger** [primefaces/community#1800] [PrimeNG] [answered]
All the examples I find for the Prime Skeleton component just place a skeleton into the page with fixed dimensions. Example:
`<p-skeleton width="10rem" styleClass="mb-2"></p-skeleton>`
But what I actually want is a skeleton that claims the space of the component that is actually loaded.
Just as React MUI does it : [React Material Skeleton - Inferring dimensions](https://mui.com/material-ui/react-skeleton/#inferring-dimensions)
How to do it?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

As an example, 

        <div class="card" style="height: 30rem">
            <p-skeleton width="100%" height="100%"></p-skeleton>
        </div>

        <div class="card" style="height: 30rem">
            <p-skeleton [style]="{ height: '100%' }"></p-skeleton>
        </div>
```

You can set the height of the wrapper.
